controllers/storagecluster/provider_server.go

```diff
@@ -62,10 +62,16 @@ func (o *ocsProviderServer) ensureDeleted(r *StorageClusterReconciler, instance
 		return reconcile.Result{}, err
 	}
 
+	serviceType := corev1.ServiceTypeClusterIP
+	if instance.Spec.ProviderAPIServerServiceType != nil {
+		serviceType = *instance.Spec.ProviderAPIServerServiceType
+	} else if instance.Spec.HostNetwork {
+		serviceType = corev1.ServiceTypeNodePort
+	}
```

Comment:
Why do we need this logic inside ensure deleted?

Reply:
removed.